lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -843,7 +843,7 @@ SECOND_PASS:
             if (!endSeg)
             {
                 // end is beyond the length of the array
-                Assert(endIndex == this->length - 1);
+                Assert(endIndex == (current->left + current->length - 1));
```

Comment:
`this->length` refers to the length field of the array, which can be anything really. What we mean to check here is that the end index points to the end of the current segment.